pages/jobs.tsx

```diff
@@ -0,0 +1,70 @@
+// import Link from 'next/link'
+import { useState, useEffect } from 'react'
+import { GetStaticProps } from 'next'
+import { siteConfig } from '@/const/meta'
+import Content from 'components/Layout/Content'
+import { Section, Title, SubTitle, TitleSmall, LinkContainer, LoadingText } from 'const/styles/pages/content'
+import { ExternalLink } from '@/const/styles/global'
+import SVG from 'react-inlinesvg'
+
+async function getJobs() {
+  const { api, deptID } = siteConfig.greenhouse
+  const getJobs = await fetch(api)
+  const { jobs } = await getJobs.json()
+  const jobsData = []
+  await jobs.map((job) => {
+    const dept = job.departments[0]
+    if (dept) {
+      dept.id === deptID ? jobsData.push(job) : null
+    }
+  })
+
+  return jobsData
+}
+
+export default function Jobs({ jobsData }) {
+  const discordURL = siteConfig.social.discord.url
+  const [jobs, setJobs] = useState(jobsData)
+  const [isLoading, setLoading] = useState(false)
+
+  // Fetch the latest jobs to possibly override static rendered (getStaticProps) SSR jobs
+  useEffect(() => {
+    setLoading(true)
+    getJobs()
+      .then((data) => {
+        setJobs(data)
+        setLoading(false)
+      })
```

Comment:
Could you add a `catch` for error handling?
Otherwise I think if the api call fails the page will crash.

Reply:
Will add that in. I think if it catches it at least should return an empty array?

Reply:
Return doesn't make sense because nothing is waiting on `getJobs` return.
What you could do is to log the exception and `setJobs([])`

Reply:
Thank you. Added the catch to log the exception. I won't setJobs([]) as the function is initially used for the `getStaticProps` to build the static content. On client side page load it run this function again (to make sure the latest jobs are fetched). In case of an error and us setting `jobs` to and empty Array, could potentially override the otherwise perfectly fine static rendered jobs.